lib/pbench/test/unit/server/database/test_server_config_db.py

```diff
@@ -1,56 +1,309 @@
+    def test_construct(self):
         """Test server config parameter constructor"""
         config = ServerConfig(key="dataset-lifetime", value="2")
         assert config.key == "dataset-lifetime"
         assert config.value == "2"
         assert str(config) == "dataset-lifetime: '2'"
+        self.check_session()
 
-    def test_create(self, db_session):
+    def test_create(self):
         """Test server config parameter creation"""
         config = ServerConfig.create(key="dataset-lifetime", value="2")
         assert config.key == "dataset-lifetime"
         assert config.value == "2"
         assert str(config) == "dataset-lifetime: '2'"
+        self.check_session(committed=[FakeRow(id=1, key="dataset-lifetime", value="2")])
 
-    def test_construct_duplicate(self, db_session):
+    def test_construct_duplicate(self):
         """Test server config parameter constructor"""
         ServerConfig.create(key="dataset-lifetime", value=1)
+        self.session.raise_on_commit = IntegrityError(
+            statement="", params="", orig=FakeDBOrig("UNIQUE constraint")
+        )
         with pytest.raises(ServerConfigDuplicate) as e:
             ServerConfig.create(key="dataset-lifetime", value=2)
         assert str(e).find("dataset-lifetime") != -1
+        self.check_session(
+            committed=[FakeRow(id=1, key="dataset-lifetime", value="1")],
+            added=[FakeRow(id=None, key="dataset-lifetime", value="2")],
+            rolledback=1,
+        )
+
+    def test_construct_null(self):
+        """Test server config parameter constructor"""
+        self.session.raise_on_commit = IntegrityError(
+            statement="", params="", orig=FakeDBOrig("NOT NULL constraint")
+        )
+        with pytest.raises(ServerConfigNullKey):
+            ServerConfig.create(key="dataset-lifetime", value=2)
+        self.check_session(
+            added=[FakeRow(id=None, key="dataset-lifetime", value="2")],
+            rolledback=1,
+        )
 
-    def test_get(self, db_session):
+    def test_get(self):
         """Test that we can retrieve what we set"""
         config = ServerConfig.create(key="dataset-lifetime", value="2")
         result = ServerConfig.get(key="dataset-lifetime")
         assert config == result
+        self.check_session(
+            committed=[FakeRow(id=1, key="dataset-lifetime", value="2")], queries=1
+        )
 
-    def test_update(self, db_session):
+    def test_update(self):
+        """Test that we can update an existing configuration setting"""
         config = ServerConfig.create(key="dataset-lifetime", value="2")
         config.value = "5"
         config.update()
+        self.check_session(committed=[FakeRow(id=1, key="dataset-lifetime", value="5")])
         result = ServerConfig.get(key="dataset-lifetime")
         assert config == result
+        self.check_session(
+            committed=[FakeRow(id=1, key="dataset-lifetime", value="5")], queries=1
+        )
```

Comment:
Previously, we needed to do the `get()` in order to confirm the behavior of the `update()`; now, that's no longer the case, so lines 296-300 are redundant (e.g., they basically duplicate lines 284-288).  What I think is missing is a call to `check_session()` between the call to `create()` and the call to `update()` (since you have one after each of `update()` and `get()`).